lib/openstudio-standards/standards/Standards.Model.rb

```diff
@@ -6114,8 +6157,79 @@ def model_get_unmet_load_hours(model)
+  # For zones works as plenum but not correctly categorized in the OS will not be identified by this function
+  # @param [OpenStudio::Model::Space] space
+  # @return boolean true if it is plenum, else false.
+  def is_space_plenum?(space)
+    # Get the zone this space is inside
+    zone = space.thermalZone
+    # the zone is a return air plenum
+    space.model.getAirLoopHVACReturnPlenums.each do |return_air_plenum|
+      if return_air_plenum.thermalZone.get.name.to_s == zone.get.name.to_s
+        # Determine if residential
+        return true
+      end
+    end
+    # the zone is a supply plenum
+    space.model.getAirLoopHVACSupplyPlenums.each do |supply_air_plenum|
+      if supply_air_plenum.thermalZone.get.name.to_s == zone.get.name.to_s
+        return true
+      end
+    end
+    # None match, return false
+    return false
+  end
+
   private
 
+  # For 2019, it is required to adjusted wwr based on building categories for all other types
+  #
+  # @param wwr_limit [Float] wwr_limit
+  # @param bat [String] building category
+  # @param wwr_list [Array] list of wwr that contains different building categories - residential, nonresidential and semiheated
+  # @return wwr_limit [Float] return adjusted wwr_limit
+  def adjust_wwr_based_on_bat(wwr_limit, bat, wwr_list)
+    return wwr_limit
+  end
+
+  # Adjust the fenestration area to the values specified by the reduction value in a surface
+  #
+  # @param surface [OpenStudio::Model:Surface] openstudio surface object
+  # @param reduction [Float] ratio of adjustments
+  # @param model [OpenStudio::Model::Model] openstudio model
+  # @return [Bool] return true if successful, false if not
+  def model_adjust_fenestration_in_a_surface(surface, reduction, model)
+    # Subsurfaces in this surface
+    # Default case only handles reduction
+    if reduction < 1.0
+      surface.subSurfaces.sort.each do |ss|
+        next unless ss.subSurfaceType == 'FixedWindow' || ss.subSurfaceType == 'OperableWindow' || ss.subSurfaceType == 'GlassDoor'
+
+        if sub_surface_vertical_rectangle?(ss)
+          sub_surface_reduce_area_by_percent_by_raising_sill(ss, reduction)
+        else
+          sub_surface_reduce_area_by_percent_by_shrinking_toward_centroid(ss, reduction)
+        end
+      end
+    end
+    return true
+  end
+
+  # Readjusted the WWR for surfaces previously has no windows to meet the
+  # overall WWR requirement.
+  # This function shall only be called if the maximum WWR value for surfaces with fenestration is lower than 90% due to
+  # accommodating the total door surface areas
+  #
+  # @param residual_ratio: [Float] the ratio of residual surfaces among the total wall surface area with no fenestrations
+  # @param space [OpenStudio::Model:Space] a space
+  # @param model [OpenStudio::Model::Model] openstudio model
+  # @return [Bool] return true if successful, false if not
+  def readjust_surface_wwr(residual_ratio, space, model)
+    return true
+  end
+
```

Comment:
This is a good question. Lot times it is hard to know when to make a function public / private. For these two functions, I put them in private primarily because:

1. incomplete logic -> The `model_adjust_fenestration_in_a_surface` does not take care of reduction ratio >=0 case, this is because the original code has not considered these cases.
2. Arguments are specific - residual_ratio is the ratio of additional surfaces need to be added to the total net wall surface area. This function mainly handles when the total fenestration surface is over wall surface area, so we need to relocate some of the additional fenestration areas to those net wall surfaces.
3. Only used in one routine - the readjust_surface_wwr only used in 90.1 PRM.

However, it does not mean these two function should stay private. I would be happy to hear any use cases that user may found these two functions useful to access.